cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ProgramLifecycleServiceTest.java

```diff
@@ -187,7 +187,11 @@ public void testProfileProgramTypeRestrictions() throws Exception {
         Optional<ProfileId> opt = SystemArguments.getProfileIdFromArgs(NamespaceId.DEFAULT,
                                                                        options.getArguments().asMap());
         Assert.assertTrue(opt.isPresent());
-        Assert.assertEquals(ProfileId.NATIVE, opt.get());
+        if (programId.getType() == ProgramType.MAPREDUCE || programId.getType() == ProgramType.SPARK) {
```

Comment:
You can use `SystemArguments. isProfileAllowed(programId.getType())`, so that in future if we allow other types, this test will not break.

Reply:
IMHO it's better to have tests be as explicit as possible. This is a rather trivial case, but what if in one of the usages of SystemArguments.isProfileAllowed we actually didn't want to add a new type? Thus the test breaking is a feature rather than a bug

Reply:
To make the intention of the test clearer, you can do something like this:

```java
Set<ProgramType> allowCustomProfiles = EnumSet.of(ProgramType.MAPREDUCE, ProgramType.SPARK, ProgramType.WORKFLOW);
for (ProgramId programId : programIds) {
  ...
  if (allowCustomProfiles.contains(programId.getType()) {
    Assert.assertEquals(profileId, opt.get());
  } else {
    Assert.assertEquals(ProfileId.NATIVE, opt.get());
  }
}
```

Reply:
Done